quic/s2n-quic-core/src/path/mtu.rs

```diff
@@ -216,29 +216,67 @@ impl_mtu!(MaxMtu, DEFAULT_MAX_MTU);
 impl_mtu!(InitialMtu, DEFAULT_INITIAL_MTU);
 impl_mtu!(BaseMtu, DEFAULT_BASE_MTU);
 
+#[non_exhaustive]
 #[derive(Debug, Eq, PartialEq)]
-pub struct MtuError;
+pub enum MtuError {
+    Validation,
+}
 
 impl Display for MtuError {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "MTU must have {} <= base_mtu (default: {}) <= initial_mtu (default: {}) <= max_mtu (default: {})",
-            MINIMUM_MTU, DEFAULT_BASE_MTU.0, DEFAULT_INITIAL_MTU.0, DEFAULT_MAX_MTU.0
-        )
+        match self {
+            MtuError::Validation => {
+                write!(
+                    f,
+                    "MTU must have {} <= base_mtu (default: {}) <= initial_mtu (default: {}) <= max_mtu (default: {})",
+                    MINIMUM_MTU, DEFAULT_BASE_MTU.0, DEFAULT_INITIAL_MTU.0, DEFAULT_MAX_MTU.0
+                )
+            }
+        }
     }
 }
 
 #[cfg(feature = "std")]
 impl std::error::Error for MtuError {}
 
+/// Information about the connection that may be used when generating or
+/// validating MTU configuration.
+pub struct ConnectionInfo<'a> {
+    pub remote_address: event::api::SocketAddress<'a>,
+    pub endpoint_config: Config,
+}
+
+impl<'a> ConnectionInfo<'a> {
+    #[inline]
+    #[doc(hidden)]
+    pub fn new(remote_address: &'a inet::SocketAddress, endpoint_mtu_config: Config) -> Self {
+        ConnectionInfo {
+            remote_address: remote_address.into_event(),
+            endpoint_config: endpoint_mtu_config,
+        }
+    }
+}
+
+/// Creates MTU config for the given connection.
+pub trait Configurator: 'static + Send {
```

Comment:
Anyone have a better name recommendation?

Reply:
I think we usually just call it `Endpoint`